lib/kafo/base_context.rb

```diff
@@ -0,0 +1,37 @@
+module Kafo
+  class BaseContext
+    def facts
+      self.class.facts
+    end
+
+    private
+
+    def self.symbolize(data)
+      case data
+      when Hash
+        {}.tap do |h|
```

Comment:
I don't insist but this 3 lines could be written as

```ruby
Hash[data.map { |key, value| [key.to_sym, symbolize(value)] }]

Reply:
My lack of Ruby experience is showing. Ty, that's a pattern I wanted to have but didn't know how :)